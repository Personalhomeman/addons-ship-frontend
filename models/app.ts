export type App = {
  appSlug: string;
  plan: string;
  title: string;
  avatarUrl: string;
  projectType: ProjectType;
  androidErrors: string[];
  iosErrors: string[];
  colors?: {
    start?: string;
    end?: string;
  };
};

export type ProjectType =
  | 'xamarin'
  | 'ios'
  | 'osx'
  | 'macos'
  | 'android'
  | 'cordova'
  | 'ionic'
  | 'react-native'
  | 'flutter'
  | 'other';
