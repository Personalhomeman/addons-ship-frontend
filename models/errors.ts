export class RequestError extends Error {
  constructor(public status: number, public statusText: string) {
    super(`${status}: ${statusText}`);
  }
}
