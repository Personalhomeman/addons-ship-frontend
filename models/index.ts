import { NextContext } from 'next';
import { Store } from 'redux';
import { Request } from 'express';

export type AppVersionProps = {
  appSlug: string;
  versionId: string;
};

export const AppVersionPageTabs = ['details', 'devices', 'test', 'activity'];

export type AppVersionPageQuery = AppVersionProps & {
  isPublic?: string;
  selectedTab?: typeof AppVersionPageTabs[number];
};

export type AppPageQuery = {
  appSlug: string;
};

export type AppVersionList = Array<{
  version: string;
  appVersions: AppVersion[];
}>;

export type AppVersion = {
  id: number;
  version: string;
  platform: string;
  buildNumber: number;
  buildSlug: string;
  lastUpdate: Date;
  description: string;
  whatsNew: string;
  minimumOs: string;
  minimumSdk: string;
  packageName: string;
  certificateExpires: Date;
  bundleId: string;
  size: number;
  supportedDeviceTypes: string[];
  distributionType: string;
  iconUrl: string;
  appName: string;
  promotionalText: string;
  keywords: string;
  reviewNotes: string;
  supportUrl: string;
  marketingUrl: string;
  scheme: string;
  configuration: string;
  publicInstallPageURL: string;
};

export interface PageContext extends NextContext {
  store: Store;
  isServer: boolean;
  req: Request;
}

export type Platform = 'ios' | 'android';
