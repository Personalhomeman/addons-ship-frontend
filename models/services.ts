export type APIConfig = {
  url: string;
};

export type AppConfig = {
  baseUrl: string;
};
