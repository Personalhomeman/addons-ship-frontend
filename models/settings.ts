export const AppSettingsPageTabs = ['general', 'notifications'];

export type AppSettingsPageQuery = {
  appSlug: string;
  selectedTab?: typeof AppSettingsPageTabs[number];
};

export type IosSettings = {
  artifactExposingWorkflows: string;
  appleDeveloperAccountEmail: string;
  appSku: string;
  appSpecificPassword: string;
  selectedProvProfile: any;
  selectedCertificate: any;
};

export type AndroidSettings = {
  artifactExposingWorkflows: string;
  track: string;
  selectedKeystoreFile: any;
  selectedServiceAccountJsonFile: any;
};

export type ProvProfile = { name: string };
export type Certificate = { name: string };
export type KeystoreFile = { name: string };
export type ServiceAccountJsonFile = { name: string };

export type Settings = {
  iosSettings?: IosSettings;
  androidSettings?: AndroidSettings;
  provProfiles?: ProvProfile[];
  certificates?: Certificate[];
  keystoreFiles?: KeystoreFile[];
  serviceAccountJsonFiles?: ServiceAccountJsonFile[];
  projectType:
    | 'xamarin'
    | 'ios'
    | 'osx'
    | 'macos'
    | 'android'
    | 'cordova'
    | 'ionic'
    | 'react-native'
    | 'flutter'
    | 'other';
};

export type AppContact = {
  id: string;
  email: string;
  isConfirmed: boolean;
  notificationPreferences: {
    newVersion: boolean;
    successfulPublish: boolean;
    failedPublish: boolean;
  };
  isMarkedForDelete?: boolean;
  isMarkedForUpdate?: boolean;
};
