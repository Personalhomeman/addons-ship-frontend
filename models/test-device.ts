export type TestDevice = {
  deviceId: string;
  deviceType: string;
  owner: string;
};
