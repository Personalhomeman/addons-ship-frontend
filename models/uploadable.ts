export type Uploadable = {
  filename: string;
  filesize: number;
  deviceType: string;
  screenSize: string;
  uploadUrl?: string;
};
