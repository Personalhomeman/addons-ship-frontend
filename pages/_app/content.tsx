import React, { Fragment, ReactNode } from 'react';
import { connect } from 'react-redux';

import Header from '@/components/Header';
import { RequestError } from '@/models/errors';

import { RootState } from '../../store';
import ErrorPage from '../_error';

type Props = {
  error?: Error | RequestError | null;
  children: ReactNode;
};

export const AppContent = ({ error, children }: Props) => {
  if (error) {
    return (
      <Fragment>
        <Header hideBreadcrumbs />
        <ErrorPage statusCode={(error as RequestError).status || 500} />
      </Fragment>
    );
  }

  return <Fragment>{children}</Fragment>;
};

const mapSateToProps = ({ error }: RootState) => ({
  error
});

// @ts-ignore
export default connect(mapSateToProps)(AppContent);
