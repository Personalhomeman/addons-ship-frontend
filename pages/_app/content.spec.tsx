import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { RequestError } from '@/models/errors';

import { AppContent } from './content';

describe('AppContent', () => {
  it('renders correclty without errors', () => {
    const tree = shallowToJson(shallow(<AppContent>Boaty McBoatface</AppContent>));
    expect(tree).toMatchSnapshot();
  });

  it('renders an error page', () => {
    const tree = shallowToJson(shallow(<AppContent error={new RequestError(404, 'Not Found')}>hidden</AppContent>));
    expect(tree).toMatchSnapshot();
  });
});
