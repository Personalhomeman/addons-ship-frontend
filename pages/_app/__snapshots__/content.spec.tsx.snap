// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContent renders an error page 1`] = `
<Fragment>
  <Connect(Header) />
  <Error
    statusCode={404}
  />
</Fragment>
`;

exports[`AppContent renders correclty without errors 1`] = `
<Fragment>
  Boaty McBoatface
</Fragment>
`;
